pkg/eventrecorder/eventrecorder.go

```diff
@@ -11,55 +11,127 @@ import (
 
 	"github.com/filecoin-project/lassie/pkg/events"
 	"github.com/filecoin-project/lassie/pkg/types"
-	logging "github.com/ipfs/go-log/v2"
-	"github.com/multiformats/go-multicodec"
+	"github.com/ipfs/go-log/v2"
```

Comment:
There is a big change set here, and I'm not clear what the benefit is. Do we believe there is a performance bottleneck on EventRecorder? What is that bottleneck? What is changed here that we believe unblocks that bottleneck? Do you have a comparable benchmark for the old code?

Reply:
It's ~half the number of allocations, at least ran on my machine, with a measurable difference in the the ns/op runtime. I added the two benchmark and perf commits before the final one that added the aggregated event recorder so that the benchmark can be ran before and after. It also continues to pass all the same tests. I'm not tied to keeping this, I had just done the work trying to make the aggregated recorder and ended up with something that was similar but benchmarked to be faster with less allocations, so I didn't want to necessarily throw it away.